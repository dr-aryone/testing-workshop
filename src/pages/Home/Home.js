import React from "react";

const Home = () => {
  return (
    <div data-testid="homeContainer">
      <h2>Homepage</h2>
    </div>
  );
};

export default Home;
