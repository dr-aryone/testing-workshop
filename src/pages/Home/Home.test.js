import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";
import Home from "./Home";

describe("Home Page", () => {
  it("Renders without crashing", () => {
    ReactDOM.render(<Home />, document.createElement("div"));

    ReactDOM.unmountComponentAtNode(document.createElement("div"));
  });

  it("Loads component", () => {
    const { getByTestId } = render(<Home />);

    const home = getByTestId("homeContainer");

    expect(home).toBeInTheDocument();
  });

  it("Match Snapshot", () => {
    const { firstChild } = render(<Home />);

    expect(firstChild).toMatchSnapshot();
  });
});
