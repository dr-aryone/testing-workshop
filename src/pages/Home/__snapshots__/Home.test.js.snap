// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Match Snapshot 1`] = `undefined`;
